pymeasure/instruments/feeltech/fy3200s.py

```diff
@@ -0,0 +1,232 @@
+from pymeasure.instruments import Instrument, SCPIUnknownMixin, Channel
+from pymeasure.adapters import SerialAdapter
+from pymeasure.instruments.validators import strict_discrete_range, strict_discrete_set
+
+
+def step_validator(step):
+    """Creates a strict_discrete_range with the specified step"""
+    return lambda x, y: strict_discrete_range(x, y, step)
+
+
+class MainChannel(Channel):
+    WAVEFORMS = {
+        "SINE": 0,
+        "SQUARE": 1,
+        "PULSE": 2,
+        "TRIANGLE": 3,
+        "SAWTOOTH_RISE": 4,
+        "SAWTOOTH_FALL": 5,
+        "DC": 6,
+        "PULSE_LORENTZ": 7,
+        "MULTITONE": 8,
+        "RANDOM_NOISE": 9,
+        "ELECTROCARDIOGRAM": 10,
+        "PULSE_TRAPEZ": 11,
+        "PULSE_SINC": 12,
+        "PULSE_NARROW": 13,
+        "GAUSSIAN_NOISE": 14,
+        "AMP_MOD": 15,
+        "FREQ_MOD": 16,
+        "ARB1": 17,
+        "ARB2": 18,
+        "ARB3": 19,
+    }
+
+    SWEEPMODES = {"LINEAR": 0, "LOG": 1}
+
+    waveform = Instrument.setting(
+        "{ch}w%i",
+        "Set the waveform of the channel.",
```

Comment:
Maybe hint at the Waveforms dictionary, such that a user knows, which values are available.